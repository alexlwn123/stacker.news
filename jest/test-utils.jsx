import React, { useCallback } from 'react'
import { render } from '@testing-library/react'
import { MockedProvider as MockedApolloProvider } from '@apollo/client/testing'
import { PriceProvider } from '../components/price'
import { MeProvider } from '../components/me'
import { LoggerProvider } from '../components/logger'
import { ChainFeeProvider } from '../components/chain-fee.js'
// import { WebLNProvider } from '../components/webln'
import { BlockHeightProvider } from '../components/block-height'
import { ServiceWorkerProvider } from '../components/serviceworker'
import { LightningProvider } from '../components/lightning'
// import ErrorBoundary from '../components/error-boundary'
import { ToastProvider } from '../components/toast'
import { ShowModalProvider } from '../components/modal'
import { ErrorBoundary } from 'react-error-boundary'
import * as lib from '@testing-library/react'

import mocks from '../__mocks__/apollo-mocks'

/** @param {{children: React.ReactNode}} */
const AllTheProviders = ({ children }) => {
  const handleError = useCallback((error, info) => {
    console.warn('Component Error Caught', error, info)
  }, [])
  const handleProviderError = useCallback((error, info) => {
    console.warn('Provider Error Caught', error, info)
  }, [])
  // const ssrData = {}
  const me = {}
  const price = {}
  const blockHeight = {}
  const chainFee = {}
  return (

    <ErrorBoundary
      onError={(error, info) =>
        handleProviderError(error, info)}
      fallback={<div>Something went wrong in a Provider</div>}
    >
      <MockedApolloProvider mocks={mocks} addTypename={false}>
        <MeProvider me={me}>
          <LoggerProvider>
            <ServiceWorkerProvider>
              <PriceProvider price={price}>
                <LightningProvider>
                  <ToastProvider>
                    {/* <WebLNProvider> */}
                    <ShowModalProvider>
                      <BlockHeightProvider blockHeight={blockHeight}>
                        <ChainFeeProvider chainFee={chainFee}>
                          <ErrorBoundary
                            onError={(error, info) =>
                              handleError(error, info)}
                            fallback={<div>Something went wrong in the component</div>}
                          >
                            {/* <Component ssrData={ssrData} {...otherProps} /> */}

                            {children}
                            {/* <PWAPrompt copyBody='This website has app functionality. Add it to your home screen to use it in fullscreen and receive notifications. In Safari:' promptOnVisit={2} /> */}
                          </ErrorBoundary>
                        </ChainFeeProvider>
                      </BlockHeightProvider>
                    </ShowModalProvider>
                    {/* </WebLNProvider> */}
                  </ToastProvider>
                </LightningProvider>
              </PriceProvider>
            </ServiceWorkerProvider>
          </LoggerProvider>
        </MeProvider>
      </MockedApolloProvider>
    </ErrorBoundary>
  )
}

/** @returns {import('@testing-library/react').render} */
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

module.exports = {
  ...lib,
  render: customRender
}
