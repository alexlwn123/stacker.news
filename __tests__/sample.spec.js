import { render, screen } from 'test-utils'
import Home from '../pages/~'
import { waitFor } from '@testing-library/react'

describe('Sample Test', () => {
  it('should be true', () => {
    expect(true).toBe(true)
  })

  it('should render "Hello World"', () => {
    render(<div>Hello World</div>)
    expect(screen.getByText('Hello World')).toBeInTheDocument()
  })

  it('should create snapshot', () => {
    const { container } = render(<div>Hello World</div>)
    expect(container).toMatchSnapshot('Hello World snapshot should render')
  })

  it('should render the loading screen', async () => {
    const { container } = render(<Home />)
    const loading = await waitFor(() => screen.getByTestId('page-loading'))
    expect(loading).toBeInTheDocument()
    expect(container).toMatchSnapshot('Loading Screen should render')
  })
})
