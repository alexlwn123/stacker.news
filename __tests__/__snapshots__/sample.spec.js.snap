// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Test should create snapshot: Hello World snapshot should render 1`] = `
<div>
  <div
    class="toast-container bottom-0 end-0 position-fixed pb-3 pe-3 toastContainer"
  />
  <div>
    Hello World
  </div>
</div>
`;

exports[`Sample Test should render the loading screen: Loading Screen should render 1`] = `
<div>
  <div
    class="toast-container bottom-0 end-0 position-fixed pb-3 pe-3 toastContainer"
  />
  <div
    class="d-flex justify-content-center mt-3 mb-1"
    data-testid="page-loading"
  >
    mock file
  </div>
</div>
`;
